### DISCUSSION TITLE
Customizable Response header?

### DISCUSSION THREAD

**Discussion by cefoot** [rejetto/hfs#279] [General]
Hello,

I'm trying to serve a WebGL Webassembly app using HFS. The files are compressed to reduce traffic. Unfortunately, the framework I'm using expects all *.gz files to be served with the Response header Content-Encoding: gzip can I configure this with HFS? Or would I be able to build a Plugin for this?

**Comment by rejetto** [comment]
sure.
https://github.com/rejetto/hfs/blob/main/dev-plugins.md
you'll find several examples in other plugins.
Create your gz/plugin.js and expose middleware()
```js
exports.init = api => ({
  middleware(ctx) { 
    if (ctx.path.endsWith('.gz'))
      ctx.set('content-encoding', 'gzip')
  }
})
```

it's just a stub, i didn't test it.
This case is so simple that could be even wrote like this, because doesn't need the api object

```js
exports.middleware = ctx => { 
    if (ctx.path.endsWith('.gz'))
      ctx.set('content-encoding', 'gzip')
}
```

**Reply by cefoot** [reply]
works exactly as I wanted it to ;) thanks!

**Comment by rejetto** [comment]
would it be nice if you had an editor in the admin panel where you simply write the code?

**Comment by cefoot** [comment]
That would make it much easier for me ;)
